tools/build_defs/repo/jvm.bzl

```diff
@@ -136,7 +136,7 @@ def _jvm_import_external(repository_ctx):
         "",
         "alias(",
         "    name = \"%s\"," % extension,
-        "    actual = \"@%s\"," % repository_ctx.name,
+        "    actual = \"//:%s\"," % name,
```

Comment:
I think this unintentionally changed the target of the alias.

For given
```
java_import_external(name = "foo", generated_rule_name = "bar")
```
The code used to create an alias for "foo". After this change, it generates an alias for "bar". 

As a result for java_import_external repositories that override default target by providing generated_rule_name (e.g. for annotation processors), the behavior for this target has changed.

Reply:
Cc @Wyverald

Reply:
https://github.com/bazelbuild/bazel/issues/24308

(FWIW, it was easy to workaround and reverting back the behavior could be also breaking change at this point if there is no other motivation to do so)